### PR TITLE
[sn-platform] Make proxy compatible for Istio mode

### LINKED ISSUE #852: [Private Cloud]: Make Proxy compatible for Istio mode

### PR DESCRIPTION
**Pull request by ciiiii**
Resolves #852 

### Motivation

1. For some old clients can't connect to Pulsar Broker directly with SNI Reverse Proxy([PR](https://github.com/apache/pulsar/pull/8117))

### Modifications

1. Make PulsarProxy support istio mode
2. Add gateway&virtualservice for proxy
3. Replace some proxy URL with broker URL in function-worker config

If you want to use admin API by Proxy, you need to specify the `.Values.image.proxy.tag` to `2.8.4.0-rc1`, because of this [bug fix](https://github.com/apache/pulsar/pull/16342).

**Test cases:**
Service:
```yaml
istio:
  enabled: true
tls:
  enabled: true
  proxy:
    enabled: true
    certSecretName: proxy-tls
ingress:
  proxy:
    enabled: true
    type: LoadBalancer
    external_domain: "proxy-service.example.com"
```
IngressGateway:
```yaml
istio:
  enabled: true
  gateway:
    selector:
      cloud.streamnative.io/role: istio-ingressgateway
    tls:
      mode: SIMPLE
      certSecretName: "istio-ingressgateway-tls"
ingress:
  proxy:
    enabled: true
    type: IstioGateway
    external_domain: "proxy-istio.example.com"
```

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Documentation

Check the box below.

Need to update docs? 

  
- [x] `no-need-doc` 
  
  (Please explain why)

### CODE REVIEW

File: charts/sn-platform/templates/proxy/proxy-cluster.yaml [hunk 1]
```diff
@@ -121,6 +123,16 @@ spec:
     imagePullSecrets:
 {{ toYaml .Values.global.imagePullSecrets | indent 6 }}
     {{- end }}
+  {{- if .Values.istio.enabled }}
```
Comment (MarvinCai) [inline charts/sn-platform/templates/proxy/proxy-cluster.yaml:126]: it doesn't seems we can configure gateway in PulsarProxy: https://github.com/streamnative/pulsar-operators/blob/master/pulsar-operator/api/v1alpha1/pulsarproxy_types.go#L85-L95

Comment (ciiiii) [reply charts/sn-platform/templates/proxy/proxy-cluster.yaml:126]: That's still in develop, I'll draft this.

**Review by fantapsody — APPROVED** [review]
Have these TLS URL changes been verified?

**Comment by ciiiii** [comment]
> Have these TLS URL changes been verified?

have verified, and it's the same as common Istio mode(without proxy).

**Comment by tuteng** [comment]
@MarvinCai PTAL

**Pull request merged**